clap_complete/src/dynamic/completer.rs

```diff
@@ -386,9 +388,16 @@ fn complete_arg_value(
                 values.extend(complete_path(value_os, current_dir, |_| true));
             }
         }
+
         values.sort();
     }
 
+    let value_os = match value {
+        Ok(value) => OsStr::new(value),
+        Err(value_os) => value_os,
+    };
+    values.extend(complete_custom_arg_value(value_os, arg));
```

Comment:
Are custom completers mutually exclusive or additive with other completion methods?

Reply:
e.g. possible values and value hint are mutually exclusive currently

Reply:
> Are custom completers mutually exclusive or additive with other completion methods?

I have also thought about this problem. I am not sure whether custom completer and ValueHint should be mutually exclusive. At first, I put the custom completer in the else branch (at the same level as the value hint) to make it mutually exclusive with possible_values.

Reply:
I lean towards being mutually exclusive, at least for now.

Reply:
There is a safe-ish fallback when `arg.get_value_hint()` is `None`. How about putting custom completions together with it?